sources/settings-extensions/host-containers/src/lib.rs

```diff
@@ -0,0 +1,119 @@
+/// host-containers settings allow users to configure multiple host containers
+use bottlerocket_settings_sdk::{GenerateResult, SettingsModel};
+use model_derive::model;
+use modeled_types::{Identifier, Url, ValidBase64};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::collections::HashMap;
+use std::convert::Infallible;
+
+/// HostContainersSettingsV1 serializes and deserializes as a
+/// HashMap<Identifier, HostContainer>.
+#[derive(Debug, Default, PartialEq)]
+pub struct HostContainersSettingsV1 {
+    pub host_containers: HashMap<Identifier, HostContainer>,
+}
+
+impl Serialize for HostContainersSettingsV1 {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.host_containers.serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for HostContainersSettingsV1 {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let host_containers = HashMap::deserialize(deserializer)?;
+        Ok(Self { host_containers })
+    }
+}
+
+#[model(impl_default = true)]
```

Comment:
Same question I had before. What are we using the `model` macro for? Can we do whatever it's doing manually instead. I think all it is doing is making these `Option`s and implementing `Serialize` and `Deserialize`. This would be better for readability if it were done manually.